web/styles/search.css

```diff
@@ -2,150 +2,232 @@
         padding: 0;
-        padding-right: 40px;
         border: none;
+        outline: none;
         border-radius: 0;
         font-family: "Source Sans 3 VF", sans-serif;
-        font-weight: 300;
-        line-height: var(--header-height);
+        font-weight: 400;
+        height: 28px;
+        /* override bootstrap min-height for input[type="text"] */
+        min-height: 28px;
+        background: transparent;
         text-overflow: ellipsis;
         overflow: hidden;
         white-space: nowrap;
-        padding-left: 35px;
-        color: hsl(0deg 0% 33%);
-
-        &:focus {
-            outline: 0;
-        }
-
-        @media (width < $sm_min) {
-            vertical-align: text-bottom;
-        }
+        flex-grow: 1;
     }
 
-    #searchbox-input-container:focus {
-        box-shadow: inset 0 0 0 2px hsl(204deg 20% 74%);
+    #searchbox-input-container {
+        display: flex;
+        align-items: center;
+        /* override .input-append style from app_components.js */
+        letter-spacing: normal;
+        height: 28px;
     }
 
-    .search_close_button {
-        position: absolute;
-        right: 0;
-        top: 6px;
-        background: none;
-        border-radius: 0;
-        border: none;
-        height: 30px;
-        text-align: center;
-        padding: 4px;
-        color: inherit;
-        opacity: 0.5;
-        font-size: 18px;
-        box-shadow: none;
-        text-shadow: none;
-        z-index: 5;
+    @media (width < $md_min) {
+        .navbar-search:not(.expanded) {
+            background: transparent;
+        }
 
-        &:hover {
+        .navbar-search:not(.expanded) .search_icon:hover {
             opacity: 1;
         }
+    }
 
-        &:disabled {
-            visibility: hidden;
+    @media (width < $sm_min) {
+        #searchbox_form {
+            margin: 0;
         }
 
-        @media (width < $sm_min) {
-            top: 0;
+        #searchbox-input-container,
+        .search-input {
+            height: var(--header-height);
         }
-    }
 
-    .search_icon {
-        position: absolute;
-        height: 100%;
-        text-decoration: none;
-        padding: 0 10px;
-        font-size: 20px;
-        z-index: 5;
-        left: 0;
-    }
-
-    #searchbox-input-container {
-        font-size: 90%;
-        letter-spacing: normal;
-        padding-left: 0;
+        #searchbox_form.expanded {
+            position: fixed;
+            left: 0;
+            right: 9px;
+            z-index: 1;
+            width: 100%;
+            border-radius: 0;
+        }
     }
-}
 
-.typeahead-menu li a {
-    .search_list_item {
+    .typeahead-menu {
         display: flex;
-        align-items: center;
+        flex-direction: column;
     }
 
-    .search_list_item .pill-container {
-        margin-left: 5px;
+    .typeahead-menu li a {
+        padding: 3px 30px;
+
+        .search_list_item {
+            display: flex;
```

Comment:
I'm not sure about this flexbox either, since `.search_list_item` appears to have a single child, `span`--maybe that's what ought to be flexed in order to solve the still-present spillover using `overflow-x: hidden`? You might also need a gap to keep "Search for" from colliding with the contents of the `<strong>` element.

Before:
<img width="605" alt="spillover" src="https://github.com/zulip/zulip/assets/170719/547a3528-658a-4c29-9ec5-626d97e71016">

After:
<img width="473" alt="contained" src="https://github.com/zulip/zulip/assets/170719/0b243b10-ffe4-4b91-93a8-b3506f543bd2">


The changes:
<img width="295" alt="flex-changes" src="https://github.com/zulip/zulip/assets/170719/51451a48-b17d-4909-9208-7556895cdea1">

Reply:
Hm, when I do that things get cut off when I don't want them to (first is your version, second is my original one)

<img width="763" alt="image" src="https://github.com/zulip/zulip/assets/5634097/09d4f379-da8b-4bb4-8184-dd6d81d179ff">

<img width="760" alt="image" src="https://github.com/zulip/zulip/assets/5634097/c7cba385-e977-4feb-965f-f5bf0ff133ba">

But I am seeing that it's still not wrapping great for that kind of input (and another I just got)

<img width="842" alt="image" src="https://github.com/zulip/zulip/assets/5634097/099f869d-b986-458f-8837-4d1cd7e117ba">

Can we pair on this together?

Reply:
Addressed during pairing!